ElrondNetwork/elrond-go
# EN-7307: transaction execution simulator

Transaction execution simulator. It will work via the API endpoint `/transaction/simulate` which receives the same request as the `/transaction/send`.
Created a new component that will simulate the execution of a transaction without affecting the state of the blockchain. In order to achieve this, some components of the new `transaction processor` were disabled or changed to read-only.

Testing steps:
1. Run a testnet with the branch's configuration (be careful that the new endpoint has to be added to the `api.toml` file)
2. Send requests to the `/transaction/simulate` endpoint. Multiple scenarios are expected:
-    move-balance transaction with invalid fields / insufficient balance / gas limit and so on. the proper error message should be returned
-    move-balance transaction with ok values. The response should signal that the transaction is correct. Please check that the receiver account hasn't updated the balance (the state was not affected). Also, in case that the gas limit was higher than the needed one, you should see a receipt inside the response
-    same two above scenarios, but for smart contract calls.
